pkg/runtime/runtime.go

```diff
@@ -409,38 +410,40 @@ func (a *DaprRuntime) initRuntime(ctx context.Context) error {
 	// Start proxy
 	a.initProxy()
 
-	// Create and start internal and external gRPC servers
-	a.daprGRPCAPI = a.getGRPCAPI()
+	// Create and start the external gRPC server
+	univAPI := a.getUniversalAPI()
+	a.daprGRPCAPI = a.getGRPCAPI(univAPI)
 
 	err = a.startGRPCAPIServer(a.daprGRPCAPI, a.runtimeConfig.apiGRPCPort)
 	if err != nil {
 		log.Fatalf("failed to start API gRPC server: %s", err)
 	}
 	if a.runtimeConfig.unixDomainSocket != "" {
-		log.Info("API gRPC server is running on a unix domain socket")
+		log.Info("API gRPC server is running on a Unix Domain Socket")
 	} else {
 		log.Infof("API gRPC server is running on port %v", a.runtimeConfig.apiGRPCPort)
 	}
 
-	a.initDirectMessaging(a.nameResolver)
-
 	// Start HTTP Server
-	err = a.startHTTPServer(a.runtimeConfig.httpPort, a.runtimeConfig.publicPort, a.runtimeConfig.profilePort, a.runtimeConfig.allowedOrigins, pipeline)
+	err = a.startHTTPServer(a.runtimeConfig.httpPort, a.runtimeConfig.publicPort, a.runtimeConfig.profilePort, a.runtimeConfig.allowedOrigins, pipeline, univAPI)
 	if err != nil {
 		log.Fatalf("failed to start HTTP server: %s", err)
 	}
 	if a.runtimeConfig.unixDomainSocket != "" {
-		log.Info("http server is running on a unix domain socket")
+		log.Info("HTTP server is running on a Unix Domain Socket")
 	} else {
-		log.Infof("http server is running on port %v", a.runtimeConfig.httpPort)
+		log.Infof("HTTP server is running on port %v", a.runtimeConfig.httpPort)
 	}
 	log.Infof("The request body size parameter is: %v", a.runtimeConfig.maxRequestBodySize)
 
+	// Start internal gRPC server (used for sidecar-to-sidecar communication)
 	err = a.startGRPCInternalServer(a.daprGRPCAPI, a.runtimeConfig.internalGRPCPort)
 	if err != nil {
 		log.Fatalf("failed to start internal gRPC server: %s", err)
 	}
-	log.Infof("internal gRPC server is running on port %v", a.runtimeConfig.internalGRPCPort)
+	log.Infof("Internal gRPC server is running on port %v", a.runtimeConfig.internalGRPCPort)
+
+	a.initDirectMessaging(a.nameResolver)
```

Comment:
Any reason `initDirectMessaging` was moved here? Otherwise LGTM

Reply:
I don't think it makes a significant difference, but the previous flow seemed weird:

1. Init gRPC API server
2. Init direct messaging
3. Init HTTP API server
4. Init internal gRPC server

It seemed best to move it to either before or after, so all servers are initialized at the same time. I landed on after for no specific reason.